### PR TITLE
feat(calendar): Adding ability to select a range of dates

### PR DESCRIPTION
**Pull request by MagicAardvark**
Adding ability to select a range of dates either fro the calendar picker or the table filters

##### **What did you change?**
modified:   demo/pages/calendar/CalendarDemo.js
new file:     demo/pages/calendar/templates/RangeDemo.html
modified:   demo/pages/table/TableDemo.js
modified:   src/elements/date-picker/DatePicker.js
modified:   src/elements/date-picker/DatePicker.spec.js
modified:   src/elements/date-picker/_DatePicker.scss
modified:   src/elements/date-picker/README.md
modified:   src/elements/dropdown/Dropdown.js
modified:   src/elements/dropdown/README.md
modified:   src/elements/table/Table.js
modified:   src/elements/table/Table.spec.js
modified:   src/elements/table/_Table.scss
modified:   src/elements/table/README.md

##### **Reviewers**
* @bvkimball @jgodi @krsween 

##### **Checklist (completed via merger)**
* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [contributing guide](https://github.com/bullhorn/novo-elements/blob/master/CONTRIBUTING.md)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Visually tested in supported browsers and devices

### CODE REVIEW

File: src/elements/date-picker/DatePicker.js [hunk 1]
```diff
@@ -131,22 +142,71 @@ export class DatePicker {
...
-
-        this.selected = day.date;
-        this.updateHeading();
-
-        if (fireEvents) {
+        if (this.range) {
+            if (!this.selected && event) {
+                this.selected = day.date;
+            } else if (this.selected && this.selected2) {
+                this.selected = day.date;
```
Comment (jgodi) [inline src/elements/date-picker/DatePicker.js]: `selected` and `selected2` is there start/end? Can they be a bit more descriptive?

Comment (MagicAardvark) [inline src/elements/date-picker/DatePicker.js]: The problem is selected is used for the normal calendar as well. I could change selected2 to selectedEnd if that makes more sense for you? I am open to ideas.

**Comment by jgodi** [comment]
Looking it over, this is the initial look for the filter on Table that I get. https://www.dropbox.com/s/s5aohgfhcr0zbt1/Screenshot%202016-08-11%2010.41.52.png?dl=0

**Comment by jgodi** [comment]
Also when toggling back and forth it does this: http://recordit.co/9I6aMjdp8T

File: src/elements/date-picker/DatePicker.js (hunk 1 above)
Comment (jgodi) [inline src/elements/date-picker/DatePicker.js]: Works for me then. No worries.

File: src/elements/table/Table.js [hunk 2]
```diff
@@ -86,10 +88,11 @@ export class NovoTableHeader {
                                 </div>
                             </div>
                             <!-- FILTER DROP-DOWN -->
-                            <novo-dropdown side="right" *ngIf="column.filtering" class="column-filters">
+                            <novo-dropdown side="right" *ngIf="column.filtering" class="column-filters" [showAfterSelect]="column.range || column.multiple">
```
Comment (jgodi) [inline src/elements/table/Table.js]: `showAfterSelect` is only a property on the item itself? Not the dropdown?

Comment (MagicAardvark) [inline src/elements/table/Table.js]: The reason for this is clicking on the custom range it stays open to reveal the calendar, other options behave the same way as any other dropdowns

**Comment by jgodi** [comment]
"Today Button" on the Range Calendar doesn't work, should it just be removed? 

https://www.dropbox.com/s/taj7zgnwo1nlp3x/Screenshot%202016-08-11%2011.39.04.png?dl=0

"Back to preset filters" should look "clickable" (cursor pointer)
https://www.dropbox.com/s/4pbn5g3jlomspdj/Screenshot%202016-08-11%2011.41.57.png?dl=0

File: src/elements/table/Table.js (hunk 2 above)
Comment (jgodi) [inline src/elements/table/Table.js]: No I mean the "input" for that property is only on the <item> component, not the <novo-dropdown>. You have it below as well on the <item> for the custom.

Comment (MagicAardvark) [inline src/elements/table/Table.js]: ah gotcha, good spot. I'll remove it.

**Pull request merged**